src/StreamJsonRpc/JsonRpc.cs

```diff
@@ -846,12 +846,14 @@ protected virtual Task<TResult> InvokeCoreAsync<TResult>(int? id, string targetN
             var requestMethodToDelegateMap = new Dictionary<string, List<MethodSignatureAndTarget>>(StringComparer.Ordinal);
             var candidateAliases = new Dictionary<string, string>(StringComparer.Ordinal);
 
+            var mapping = new MethodNameMap(target.GetType().GetTypeInfo());
+
             for (TypeInfo t = target.GetType().GetTypeInfo(); t != null && t != typeof(object).GetTypeInfo(); t = t.BaseType?.GetTypeInfo())
             {
                 foreach (MethodInfo method in t.DeclaredMethods)
                 {
-                    var attribute = (JsonRpcMethodAttribute)method.GetCustomAttribute(typeof(JsonRpcMethodAttribute));
-                    var requestName = attribute?.Name ?? method.Name;
+                    var attribute = mapping.FindAttribute(method);
```

Comment:
Should you keep this call, or should the place where you check attribute == null at the end of this method (for candidates), instead check to see if requestName == null?